lib/fog/linode/models/compute/kernels.rb

```diff
@@ -11,20 +11,27 @@ def all
           load kernels
         end
 
+        def all_kvm
+          load kernels(:isKVM => 1)
+        end
+
+        def all_xen
+          load kernels(:isXen => 1)
+        end
```

Comment:
I was thinking more about backwards compatibility in the future. It would be easier to remove an option then a method. But thats kind of my own opinion here. Not exactly something to be followed.